### PR TITLE
Add actions to transformation panel

### LINKED ISSUE #1683: Some transformation panel buttons should use actions.js, so they become "shortcuttable"
Specifically:
- the align objects buttons
- the distribute objects buttons
- the path operations (remove overlap, etc.)

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #1683

### CODE REVIEW

**Comment by justvanrossum** [comment]
We have a fundamental problem here: the actions only get registered once the _panel_ gets updated/shown. This means that if you _hide_ the panel, then reload, and shortcut does not work.

We somehow need to refactor this, but I need to think a bit more about how to best approach it.

Also: there is an existing bug where you get a console error message if you use one of the actions while there is no selection:
image

**Comment by ollimeier** [comment]
> We have a fundamental problem here: the actions only get registered once the _panel_ gets updated/shown. This means that if you _hide_ the panel, then reload, any shortcuts do not work.
> 
> We somehow need to refactor this, but I need to think a bit more about how to best approach it.
> 
> Also: there is an existing bug where you get a console error message if you use one of the actions while there is no selection: image

I am not able to reproduce the error. Can you please let me know if this error still exists after my latest changes?

**Comment by justvanrossum** [comment]
Yes, the error still exists, but only in the "remove overlaps" action/button: make sure there is _no_ selected glyph in the canvas, then push the "remove overlaps" button. (All other buttons appear to be fine in this case.)

**Comment by ollimeier** [comment]
> Yes, the error still exists, but only in the "remove overlaps" action/button: make sure there is _no_ selected glyph in the canvas, then push the "remove overlaps" button. (All other buttons appear to be fine in this case.)

Ah, I see. Thanks. I fixed it.

File: src/fontra/views/editor/panel-transformation.js [hunk 1]
```diff
@@ -101,6 +102,46 @@ export default class TransformationPanel extends Panel {
       skewY: 0,
       customDistributionSpacing: null,
     };
+    this.registerActions();
+  }
+
+  registerActions() {
+    const topic = "0070-action-topics.selection-transformations";
+
+    const registerActions = [
```
Comment (justvanrossum) [inline src/fontra/views/editor/panel-transformation.js:111]: I'd prefer a different vairable name here. I think I'd like `moveActions`.

File: src/fontra/views/editor/panel-transformation.js [hunk 2]
```diff
@@ -101,6 +102,46 @@ export default class TransformationPanel extends Panel {
...
+    ];
+    for (const [keyPart, moveDescriptor] of registerActions) {
+      registerAction(
+        `action.selection-transformation.${keyPart}`,
+        { topic, titleKey: `sidebar.selection-transformation.${keyPart}` },
+        () => this.moveObjects(moveDescriptor)
+      );
+    }
+
+    const registerActionsPathOperations = [
```
Comment (justvanrossum) [inline src/fontra/views/editor/panel-transformation.js:129]: And for this one I'd prefer `pathActions`.

**Review by justvanrossum — COMMENTED** [review]
Two minor nits, then this is done I think.

File: src/fontra/views/editor/panel-transformation.js (hunk 1 above)
Comment (ollimeier) [reply src/fontra/views/editor/panel-transformation.js]: done.

File: src/fontra/views/editor/panel-transformation.js (hunk 2 above)
Comment (ollimeier) [reply src/fontra/views/editor/panel-transformation.js]: done.

**Review by justvanrossum — APPROVED** [review]

**Pull request merged**